Mark a Java dependency as GraalVM compatible (#9117)
ballerina-platform/ballerina-dev-website: swan-lake/development-tutorials/source-code-dependencies/package-references.md

>**Note:** When the scope has been specified as `provided`, the values `groupId`, `artifactId`, and `version` will be considered mandatory fields for that dependency.

**Mark a Java dependency as GraalVM compatible**

> A Java dependency can be marked as GraalVM compatible by passing `graalvmCompatible = true` property as follows:

Reviewer: ```suggestion
A Java dependency can be marked as GraalVM compatible by passing the `graalvmCompatible = true` property as follows:
```

```toml
[[platform.java11.dependency]]
groupId = "<group-id>"
artifactId = "<artifact-id>"
version = "<version>"
graalvmCompatible = true
```

> If the package uses only GraalVM compatible Java dependencies, The package is indicated as GraalVM compatible. 

Reviewer: ```suggestion
If all the Java dependencies used in the package are marked as GraalVM compatible, the package is considered GraalVM compatible. 
```
